### ISSUE TITLE
Potential false-positive in 1.1.185 involving class-scoped generic type alias

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#2549]
**Describe the bug**
pyright-basic raises an error "Generic type alias within class cannot use bound type variables" for code that doesn't specify a generic alias withing a class.

**To Reproduce**
Although it is a bit contrived, the following is the simplest case that I can muster to reproduce the [error raised in `hydra-zen`](https://github.com/mit-ll-responsible-ai/hydra-zen/runs/4141824970?check_suite_focus=true).

```python
from typing import TypeVar, Callable, Sequence, Union, Protocol

T1 = TypeVar("T1")
T2 = TypeVar("T2", covariant=True)


class Prot(Protocol[T2]):
    pass


A = Union[Callable[[T1], T1], Prot[Callable[[T1], T1]]]
B = Union[A, Sequence[A]]  # error: Generic type alias within class cannot use bound type variables _T2
```

**Expected behavior**
The above code seems OK to me. At the very least I do not see where the "generic type alias within class" is. Replacing `B` everywhere with `Union[A, Sequence[A]]` removes the error.

**VS Code extension or command-line**
Running pyright in basic mode via commandline. Version 1.1.185.

**Comment by erictraut** [comment]
Sorry about the regression. I hadn't considered this case when implementing [this new check](https://github.com/microsoft/pyright/issues/2541).

I've introduced a fix, and it will be included in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.186, which I just published. It will also be included in the next release of pylance.
